src/java.base/unix/classes/sun/nio/fs/UnixCopyFile.java

```diff
@@ -209,8 +213,14 @@ private static void copyDirectory(UnixPath source,
             }
             done = true;
         } finally {
-            if (dfd >= 0)
-                close(dfd);
+            if (dfd >= 0) {
+                try {
+                    close(dfd);
+                } catch (UnixException e) {
+                    if (done)
+                        e.rethrowAsIOException(target);
```

Comment:
Should we just ignore the exception irrespective of whether or not it is `done`? That I think would probably be a bit more consistent with some other places where we ignore exception from `close`. I think ignoring this instead of (conditionally) throwing this will also help propagate any previous raised exception, where a few lines above this `finally` block there's an "inner" try/catch which can potentially throw an `IOException` (line 211).